mitodl/micromasters
# Added unenroll program(s) feature

#### What are the relevant tickets?
fixes https://github.com/mitodl/micromasters/issues/4056

#### What's this PR do?
It adds a wizard to unenroll from one or many programs. And update UI on unenroll.

- [x] Added api
- [x] Add unenroll card on settings
- [x] Add unenroll programs wizard
  - [x] display enrolled programs
  - [x] Add multi select programs widget
  - [x] api call
  - [x] show spinner on api call
- [x] dashboard card when no program selected
- [x] Unit tests for front end
- [x] Unit tests for backend
 
#### How should this be manually tested?
- Enroll programs one or many
- Go to settings and click unenroll link
- then select program(s)
- check dashboard
- when no program left i.e you unenroll all, check new card on dashboard with enroll link

@pdpinch 
#### Screenshots (if appropriate)
-  <img width="825" alt="screen shot 2018-08-03 at 5 25 22 pm" src="https://user-images.githubusercontent.com/10431250/43642760-41d601f0-9742-11e8-9523-8b3a85c3403f.png">
- ![screen shot 2018-08-02 at 3 07 14 pm](https://user-images.githubusercontent.com/10431250/43577621-c7687772-9665-11e8-99d4-3a315c292401.png)
-  <img width="650" alt="screen shot 2018-08-02 at 3 02 58 pm" src="https://user-images.githubusercontent.com/10431250/43577405-2faddd0a-9665-11e8-8aba-5a0fd26512c8.png">
-  <img width="550" alt="screen shot 2018-08-02 at 3 08 47 pm" src="https://user-images.githubusercontent.com/10431250/43577707-01999624-9666-11e8-8365-f9abf1a75220.png">
- ![screen shot 2018-08-03 at 4 30 50 pm](https://user-images.githubusercontent.com/10431250/43640777-a0d51964-973a-11e8-8ce9-c8fde7ded8f1.png)
